src/SearchInput/SearchInput.js

```diff
@@ -32,6 +32,11 @@ class SearchInput extends Component {
     };
 
     handleListItemClick = (event, item) => {
+        this.setState({
+            value: item.text,
+            isExpanded: false,
+            searchExpanded: false
+        });
         item.callback ? item.callback() : null;
```

Comment:
I see let calling the onChange callback after setState, we could follow the same pattern.

Also, if you are setting the value in state, it looks like you need to recompute the filteredResult, like this:

```js
        this.setState((prevState) => ({
            value: item.text,
            isExpanded: false,
            searchExpanded: false,
            filteredResult: this.filterList(prevState.searchList, item.text)
        }), () => {
            if (item.callback) {
                item.callback();
            }
        });     
```

Otherwise it can show every item in the list prefixed with the selected text

<img width="184" alt="Screen Shot 2020-04-02 at 10 42 56 AM" src="https://user-images.githubusercontent.com/5314713/78269174-d5f66080-74ce-11ea-9ad7-16ab09923949.png">

instead of

<img width="237" alt="Screen Shot 2020-04-02 at 10 44 12 AM" src="https://user-images.githubusercontent.com/5314713/78269248-f45c5c00-74ce-11ea-82eb-b41f84c99397.png">

Reply:
Hi @skvale  and thanks for the review I pushed a new fix.